src/actions/commands/actions.ts

```diff
@@ -308,8 +308,31 @@ export class CommandNumber extends BaseCommand {
 
   public async exec(position: Position, vimState: VimState): Promise<VimState> {
     const number = parseInt(this.keysPressed[0], 10);
-
-    vimState.recordedState.count = vimState.recordedState.count * 10 + number;
+    const operatorCount = vimState.recordedState.operatorCount;
+
+    if (operatorCount > 0) {
+      const lastAction =
+        vimState.recordedState.actionsRun[vimState.recordedState.actionsRun.length - 2];
+      if (!(lastAction instanceof CommandNumber)) {
```

Comment:
Is this branch necessary? Isn't `vimState.recordedState.count` 0 before it's set? (could be wrong about that)

Reply:
This is to handle the first number after an operator in a different way from the following ones. For example, if you press '3d12w' the '3' makes `vimState.recordedState.count = 3`, then 'd' makes `vimState.recordedState.operatorCount = 3`, then when pressing '1' this branch sees that the action previous to itself was not a `CommandNumber` which means it was the operator so it makes `vimState.recordedState.count = 3 * 1`, then the last number '2' goes to the other branch and makes `vimState.recordedState.count= 3 * 10 + 2 * 3` which makes the final count 36 (3*12).

Reply:
Got it 👍

Reply:
Also the `operatorCount` is only set when we get an Operator and `recordedState.count > 0`. So when `operatorCount` is higher than 0 we know that at least we had a number followed by an operator key followed by another number (example `2d3`). So this first branch handles only the first number after an operator that had a count before, the branch below handles all the following numbers and the last branch handles all the numbers before an operator and the numbers after an operator that didn't had a count before.